src/registry.rs

```diff
@@ -211,22 +212,22 @@ impl Registry {
     /// (which includes the case of re-registering the same Collector), the
     /// AlreadyReg error returns.
     pub fn register(&self, c: Box<Collector>) -> Result<()> {
-        self.r.write().unwrap().register(c)
+        self.r.write().register(c)
     }
 
     /// `unregister` unregisters the Collector that equals the Collector passed
     /// in as an argument.  (Two Collectors are considered equal if their
     /// Describe method yields the same set of descriptors.) The function
     /// returns error when the Collector is not registered.
     pub fn unregister(&self, c: Box<Collector>) -> Result<()> {
-        self.r.write().unwrap().unregister(c)
+        self.r.write().unregister(c)
     }
 
     /// `gather` calls the Collect method of the registered Collectors and then
     /// gathers the collected metrics into a lexicographically sorted slice
     /// of MetricFamily protobufs.
     pub fn gather(&self) -> Vec<proto::MetricFamily> {
-        self.r.read().unwrap().gather()
+        self.r.read().gather()
```

Comment:
Will this function last for a long time?

Reply:
It should be fast, also other methods are rarely called.